index.js

```diff
@@ -90,14 +91,32 @@ class Sqlite extends Base {
     ], cb)
   }
 
-  upsert (data, cb) {
-    const res = this._buildUpsertQuery(data)
-    this.db.run(res.query, res.data, function (err) {
-      if (err) return cb(err)
-      cb(null, { lastID: this.lastID })
+  async runAsync (sql, params = []) {
+    return new Promise((resolve, reject) => {
+      return this.db.run(sql, params, function (err, res) { // passing an arrow function won't work
+        if (err) return reject(err)
+        return resolve(this)
+      })
     })
   }
 
+  async allAsync (sql, params, ...rest) {
+    return promisify(this.db.all.bind(this.db))(sql, params, ...rest)
+  }
+
+  async getAsync (sql, params, ...rest) {
+    return promisify(this.db.get.bind(this.db))(sql, params, ...rest)
+  }
+
+  async upsert (data, cb) {
+    const { query, data: params } = this._buildUpsertQuery(data)
```

Comment:
can we use https://github.com/bitfinexcom/lib-js-util-promise/blob/main/src/resolve-promise-cb.js similar to how we used it in bfx pay?

Reply:
Sounds like a better idea

Reply:
I implemented a separate upsertAsync funciton so thie suggestion no longer needed